rfcs/experimental/flow_graph_wait_for_one/README.md

```diff
@@ -0,0 +1,481 @@
+
+It may buffer both ``t`` and ``metainfo`` or broadcast the result and the ``info`` to the successors of the node. 
+
+The existing API ``try_put_task(const T& t)`` can reuse the new one with the empty metainfo object.
+
+```cpp
+template <typename T>
+class sender {
+public:
+    // Existing API
+    virtual bool try_get(T& t) { return false; }
+    virtual bool try_reserve(T& t) { return false; }
+
+    // New API
+    virtual bool try_get(T& t, message_metainfo& info) { return false; }
+    virtual bool try_reserve(T& t, message_metainfo& info) { return false; }
+};
+```
+
+For each particular implementation of ``sender``, ``try_get`` gets the element and the metainfo from the buffer and assigns the message to ``t`` and
+the metainfo to ``info``. The reference counter/s, associated with the stored metainfo are released by ``1``.
+
+``try_reserve`` implementation reserves the element and the corresponding metainfo inside of the buffer and feels the placeholders provided. Since the elements are not
+removed from the buffer, the reference counter/s remains unchanged and would be release by ``1`` if ``try_consume`` would be called. 
+
+## Nodes behavior
+
+This chapter describes detailed behavior of each Flow Graph node when the item and the metainformation is received. Similarly to the message itself, the metainformation
+can be received from the predecessor node (explicit ``try_put_task`` call) or initially from ``try_put_and_wait``.
+
+### Queueing ``function_node``
+
+If the concurrency of the ``function_node`` is set to ``unlimited``, the node creates a task for executing the body of the node. The created task holding the reference counter on each
+``wait_context``s stored in ``metainfo`` and also wraps the ``metainfo`` object itself since it would be broadcasted to the successors when the task is completed.
+
+If the concurrency is not ``unlimited``, the call to ``try_put_task`` tries to occupy the concurrency of the node. If the thread limit is not yet reached - behaves the same as 
+in the ``unlimited`` case. Otherwise, both input message and the metainfo are stored in the internal queue of the node. When one of the node tasks is completed, it retrieves 
+the postponed message and the corresponding metainfo from the queue and spawns a task to process it.
+
+Since the ``function_node`` guarantees that all of the elements would be retrieved from the internal queue at some time, [buffering issues](#buffering-the-metainfo) cannot take place.
+
+### Rejecting ``function_node``
+
+If the concurrency of the node is set to ``unlimited``, behaves the same as in the ``queueing`` case described above.
+
+Otherwise, if the concurrency limit of the node is reached, both message and the associated metainformations would be rejected and the predecessor that called the ``try_put_task``
+is responsible on buffering both of them.
+
+If the predecessor is not the buffering node, both message and the metainfo would be lost.
```

Comment:
You should say what "lost" means in this context. How does it affect waiting.

Reply:
Updated the description